stl/inc/random

```diff
@@ -238,32 +238,60 @@ private:
     vector<result_type> _Myvec;
 };
 
+template <class _Real, size_t _Minbits, class _Result, _Result _Ix>
+_NODISCARD constexpr int _Generate_canonical_helper() {
+    if constexpr ((_Ix & (_Ix - 1)) == 0) {
+        constexpr auto _Log2 = _STD _Countr_zero(_Ix);
+        return static_cast<int>((_Minbits + _Log2 - 1) / _Log2);

```

Comment:
With `_Ix` equal to `0` we pick this branch, was this considered?

Reply:
Yes, very much so. When `_Gen::max() - _Gen::min() + 1 == _Ix == 0`, the generator generates the full range of its `result_type` - i.e., `_Gen::max() == ~result_type{0}` and `_Gen::min() == 0`. So the log<sub>2</sub> of the range is in fact the width of the `result_type`, which happens to be `_Countr_zero(result_type{0}) == _Countr_zero(_Ix)`.